browser/search_engines/search_engine_provider_service_browsertest.cc

```diff
@@ -220,14 +223,20 @@ IN_PROC_BROWSER_TEST_F(SearchEngineProviderServiceTest,
       TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_DUCKDUCKGO;
 
   // Check guest profile's search provider is set to ddg.
-  EXPECT_EQ(service->GetDefaultSearchProvider()->data().prepopulate_id,
+  EXPECT_EQ(template_service->GetDefaultSearchProvider()->data().prepopulate_id,
             provider_id);
 
   auto bing_data = TemplateURLPrepopulateData::GetPrepopulatedEngine(
       guest_profile->GetPrefs(),
       TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_BING);
   TemplateURL bing_url(*bing_data);
-  service->SetUserSelectedDefaultSearchProvider(&bing_url);
+  template_service->SetUserSelectedDefaultSearchProvider(&bing_url);
+
+  auto* search_engine_provider_service =
+      static_cast<GuestWindowSearchEngineProviderService*>(
+          SearchEngineProviderServiceFactory::GetForProfile(guest_profile));
+  search_engine_provider_service->OnTemplateURLServiceChanged();
```

Comment:
was this test broken before? after it would call `SetUserSelectedDefaultSearchProvider`, it seems it was missing the `OnTemplateURLServiceChanged`

Reply:
It was called before.
Previously, OTR's `AutocompleteClassifier` uses OTR's `TemplateURLService` via `BraveAutocompleteProviderclientForClassifier` .  and it's rollbacked with this PR.
I think OTR's `AutocompleteClassifier` was initialized after Guest Profile was created and then TemplateURLService was also initialized and loaded.
With this PR, this test is ended before OTR's `TemplateURLService` gets loaded state.
When `TemplateURLService::OnWebDataServiceRequestDone()` is called, it enters loaded state. but it's not called before finishing this test.
Of course, we can wait `TemplateURLService` gets loaded but just calling `OnTemplateURLServiceChanged()` seems sufficient for this test purpose.

Reply:
This makes sense - I see that now in the deleted code 😄 
ex: `BraveAutocompleteProviderClientForClassifier::GetTemplateURLService`

GitHub had the files collapsed so it didn't match on find 👍